x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureopenai/completion/AzureOpenAiCompletionServiceSettings.java

```diff
@@ -0,0 +1,187 @@
+
+    private final String resourceName;
+    private final String deploymentId;
+    private final String apiVersion;
+
+    private final RateLimitSettings rateLimitSettings;
+
+    public AzureOpenAiCompletionServiceSettings(
+        String resourceName,
+        String deploymentId,
+        String apiVersion,
+        @Nullable RateLimitSettings rateLimitSettings
+    ) {
+        this.resourceName = resourceName;
+        this.deploymentId = deploymentId;
+        this.apiVersion = apiVersion;
+        this.rateLimitSettings = Objects.requireNonNullElse(rateLimitSettings, DEFAULT_RATE_LIMIT_SETTINGS);
+    }
+
+    public AzureOpenAiCompletionServiceSettings(StreamInput in) throws IOException {
+        resourceName = in.readString();
+        deploymentId = in.readString();
+        apiVersion = in.readString();
+
+        if (in.getTransportVersion().onOrAfter(TransportVersions.ML_INFERENCE_RATE_LIMIT_SETTINGS_ADDED)) {
+            rateLimitSettings = new RateLimitSettings(in);
+        } else {
+            rateLimitSettings = DEFAULT_RATE_LIMIT_SETTINGS;
+        }
+    }
+
+    private AzureOpenAiCompletionServiceSettings(AzureOpenAiCompletionServiceSettings.CommonFields fields) {
+        this(fields.resourceName, fields.deploymentId, fields.apiVersion, fields.rateLimitSettings);
+    }
+
+    public String resourceName() {
+        return resourceName;
+    }
+
+    public String deploymentId() {
+        return deploymentId;
+    }
+
+    @Override
+    public RateLimitSettings rateLimitSettings() {
+        return DEFAULT_RATE_LIMIT_SETTINGS;
+    }
+
+    public String apiVersion() {
+        return apiVersion;
+    }
+
+    @Override
+    public DenseVectorFieldMapper.ElementType elementType() {
+        return DenseVectorFieldMapper.ElementType.FLOAT;
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.startObject();
+
+        toXContentFragmentOfExposedFields(builder, params);
+
+        builder.endObject();
+        return builder;
+    }
+
+    private void toXContentFragmentOfExposedFields(XContentBuilder builder, ToXContent.Params params) throws IOException {
+        builder.field(RESOURCE_NAME, resourceName);
+        builder.field(DEPLOYMENT_ID, deploymentId);
+        builder.field(API_VERSION, apiVersion);
+        rateLimitSettings.toXContent(builder, params);
+    }
+
+    @Override
+    public ToXContentObject getFilteredXContentObject() {
+        return (builder, params) -> {
+            builder.startObject();
+
+            toXContentFragmentOfExposedFields(builder, params);
+
+            builder.endObject();
+            return builder;
+        };
+    }
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return TransportVersions.ML_INFERENCE_AZURE_OPENAI_EMBEDDINGS;
```

Comment:
We'll want this to be `ML_INFERENCE_AZURE_OPENAI_COMPLETIONS`.

Reply:
[Use correct minimal supported transport version](https://github.com/elastic/elasticsearch/pull/108352/commits/146e0001e5ab65133fe4d917f799d85c86be36d1)